### ISSUE TITLE
Error while loading page lovelace. on 2024.5.0

### ISSUE THREAD

**Issue by jaymunro** [home-assistant/frontend#20697]
EDIT: If you have this issue on iOS, just don't add more "Me too" type comments but use the thumbs up on the relevant comment perhaps this top comment, or use the "Subscribe" button. If you have something new to add, then please do comment.

### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Issue exists on:
- the HA Companion app (2023.12.1) on an old ipad running iOS 12.5.7 (max for the iPad)
- Safari on an old ipad running iOS 12.5.7 (max for the iPad)
- Google Chrome on all iPads & phones running newest versions of iOS (17.4.1) and Chrome (124.0.6367.88)

Issue does **not** exist on 
- Safari on all iPads & phones running newest versions of iOS (17.4.1)
- the HA Companion app on new iPads and iPhones running iOS 17.4.1
- Chrome on MacOS (latest versions)
- All older HA versions 2024.4.x and lower.

When loading the app the login page is empty of text but has graphics and input is possible. See screenshot below.
![327292578-af4dd659-e9be-4e78-b730-9b6cb708b53a](https://github.com/home-assistant/frontend/assets/1166024/c2425bc8-38e2-49c5-9022-21f3d17fb302)

After login UI does not show and only displays an error `Error while loading page lovelace.` as per the screenshot below:
![327293641-0bc9c10c-6353-4a5c-94d0-c9c8ea1b982d](https://github.com/home-assistant/frontend/assets/1166024/51ca0de5-89c2-4cf1-b0ee-65a566ffa10d)

This occurs on all menu items on the left including "Settings".

### Describe the behavior you expected

To use the UI as per normal

### Steps to reproduce the issue

1. Upgrade to 2024.5.0
2. Try to login and use HA on an old iOS version or any Chrome browser on iOS
3. If the issue does not present itself on a new iOS and Chrome then chose the Chrome option "Request desktop site"
4. The UI will fail to display dashboards, Energy, Logbook, History, Settings (but **will** display Developer tools and addons such as HACS, ESPHome, Studio code server, etc).

### What version of Home Assistant Core has the issue?

2024.5.0

### What was the last working version of Home Assistant Core?

2024.4.4

### In which browser are you experiencing the issue with?

Google Chome 124.0.6367.88 and Home Assistant Companion 2023.12.1

### Which operating system are you using to run this browser?

iOS 17.1 & iOS 12.5.7

**Comment by gerrypc72** [comment]
Same problem, same condition

**Comment by jaymunro** [comment]
> Same problem on older iPad used for dashboard view only.

That's pretty much the use case of mine, but it is also used for control by family members for lighting, heating and radio. It's completely offline now and that's an issue in the family.

I can't revert to the backup without losing all the sensor data since the upgrade so not keen to do that either.

**Comment by oldiebastler** [comment]
Ich habe das gleiche Problem mit meinem alten iPad.

**Comment by HellJohny** [comment]
iPad mini 2. Safari browser. After update same problem.

**Comment by Olsenoli** [comment]
I habe the same Problem. Is there already a possible solution to the problem?

**Comment by jaymunro** [comment]
Does this also affect anyone else loading HA in Chrome on a recent iPad or iPhone? A dev in Discord is saying it is not affecting him, however it is broken in Chrome for me on an iPad with current iOS too.

**Comment by gerrypc72** [comment]
I have a newer iPad with iPadOS 17.4.1and it works fine on that

**Comment by jaymunro** [comment]
On the old iPad, rather than using the private browser I cleared all cache and browser history and now it results in "Safari cannot open the page because it could not connect to the server." This suggests to me that this is related to the secure connections feature that got pulled.

**Comment by jaymunro** [comment]
> I have a newer iPad with iPadOS 17.4.1and it works fine on that

Are you saying Chrome works fine? Or did you use Safari / Companion app?

**Comment by jaymunro** [comment]
Ok, this may be a big clue and help reproduce the issue. In Chrome on my recent iPad, I chose the option to "Request Mobile Site" and the issue goes away. I then "Request desktop site" and the issue returns. Same on the iPhone 12. That seems to be the difference.

Can someone else reproduce this?

**Comment by Joshbloam** [comment]
Same issue here. iPad air first gen IOS 12.5.7. Companion app 2023.12.1. Restored my HA instance back to 2024.4.0 to recover.

**Comment by patkino** [comment]
Same.issue here. Chinese android 8.1 control panel.
Companion doesn't work, web interface via Chrome works fine

**Comment by lukonin** [comment]
Same issue with iPad Mini 2. Reverted back to 2024.4.0.

**Comment by gerrypc72** [comment]
> Ok, this may be a big clue and help reproduce the issue. In Chrome on my recent iPad, I chose the option to "Request Mobile Site" and the issue goes away. I then "Request desktop site" and the issue returns. Same on the iPhone 12. That seems to be the difference.
> 
> Can someone else reproduce this?

On older iPad Air (2014) IOS 12.5.7 It doesn't work via the Companion App or the Safari browser, not even by requesting the desktop site. Both the Companion App and the Safari browser work perfectly on the iPad Pro (2018) IOS 17.4.1. I don't have Chrome

**Comment by jaymunro** [comment]
> > Ok, this may be a big clue and help reproduce the issue. In Chrome on my recent iPad, I chose the option to "Request Mobile Site" and the issue goes away. I then "Request desktop site" and the issue returns. Same on the iPhone 12. That seems to be the difference.
> > Can someone else reproduce this?
> 
> On older iPad Air (2014) IOS 12.5.7 It doesn't work via the Companion App or the Safari browser, not even by requesting the desktop site. Both the Companion App and the Safari browser work perfectly on the iPad Pro (2018) IOS 17.4.1. I don't have Chrome

Yes, same for me on the newer iPad. Companion and Safari have no issues. 
It is just Chrome desktop version on the newer iPad and everything on the older iPad.

I'm hoping others can confirm the behaviour of Chrome between desktop and mobile versions on the newer iPads. This may help the devs replicate it and fix it faster.

**Comment by pvmil** [comment]
I have the same issiue, iPad Air running IOS 12.5.7
The installed chrome and safari browsers have the login page issue, after logging in they do show the menubar. Opening dashboards, settings etc does not show anything but 'Error while loading page dashboard'
The companion app shows the setting cog, which works, but I do not get any other information after that, no menu bar, nothing.

**Comment by jaymunro** [comment]
@frenck @balloob are you aware of this rather major regression?

**Comment by piitaya** [comment]
Can someone can provide browser logs? 🙂

**Comment by pvmil** [comment]
> Can someone can provide browser logs? 🙂

I made a log using the chrome browser by using the chrome://inspect url.

The output I got:
`http://192.168.1.11:8123/config/dashboard
INFO%cBROWSER_MOD 2.3.0 IS INSTALLED %cBrowserID: 979d4e07-f610dc20 color: green; font-weight: bold
ERRORError loading page dashboard TypeError: i is not a function. (In 'i()', 'i' is an instance of Object)`

This triggers my use of [browser_mod](![image](https://github.com/home-assistant/frontend/assets/14914859/1e6985b5-383b-483b-b50c-777a43caec90)), (HACS) are others using this?

**Comment by silamon** [comment]
Normally these errors should also be logged to the system logging. Can they found there, possibly with a full stacktrace?

**Comment by pvmil** [comment]
> Normally these errors should also be logged to the system logging. Can they found there, possibly with a full stacktrace?

I do not see any related error in the logging in HA (checked 'Home assistant Core', 'supervisor', 'Host').

**Comment by silamon** [comment]
Apparently it's possible to open the browser console from chrome://inspect by using  "inspect". We'll need the stack trace of any of those: ERROR Error loading page dashboard TypeError: i is not a function. (In 'i()', 'i' is an instance of Object)`.

**Comment by samfaitmalomolet** [comment]
Hello,
I have this data from inspect on my ipad air
![20240503_194957](https://github.com/home-assistant/frontend/assets/74918326/74691a2f-db2f-48c7-b7e4-8723b4755415)

**Comment by jaymunro** [comment]
From https://developer.chrome.com/blog/debugging-chrome-on-ios/

> On your iOS device you need:
> 
> iOS 16.4 or greater.
> Chrome 115 or greater.
> On your Mac you need:
> 
> The Safari Develop menu enabled. Enable this from Safari application settings, Advanced Settings tab.
> On your iOS device launch the Chrome app and navigate to Settings. In Content Settings, enable Web Inspector. Relaunch Chrome for iOS after changing this setting.
> 
> Debugging your site
> 
> With the setting enabled, you can debug any page running in Chrome for iOS by following these steps.
> 
> Connect your iOS device to a Mac with a cable.
> In Chrome for iOS, navigate to the website which you'd like to inspect.
> On your Mac, launch Safari. From the Develop menu, hover over the name of your connected device and select the url of the website which you would like to inspect.
> The displayed Web Inspector window will allow debugging of the current web view.

I'll try this later today but if someone can beat me to it, great. I need to go out now and mow the lawns to make up for my mistake in disabling our main control panel into HA.

**Comment by jaymunro** [comment]
Ok, I was able to remotely connect to Safari with Web Inspector on my old iPad.

For the login screen where all text is missing, this is shown in the console:
```
[Error] Unhandled Promise Rejection: TypeError: i is not a function. (In 'i()', 'i' is an instance of Object)
	(anonymous function) (848.RLtFmELmVmg.js:1:621)
	(anonymous function) (848.RLtFmELmVmg.js:1:5359)
	d (authorize.bgGP9HuyRNc.js:2:450428)
	d
	promiseReactionJob
[Error] Unhandled Promise Rejection: TypeError: a is not a function. (In 'a()', 'a' is an instance of Object)
	(anonymous function) (4115.gLuPLXiiJG8.js:1:190)
	promiseReactionJob
```

After logging in, this is shown in the console when viewing my main dashboard:
```
[Error] Unhandled Promise Rejection: SyntaxError: Unexpected token '.'
	(anonymous function)
	rejectPromise
	promiseReactionJob
[Error] Unhandled Promise Rejection: TypeError: 'text/html' is not a valid JavaScript MIME type.
	(anonymous function)
	rejectPromise
	promiseReactionJob
[Error] Unhandled Promise Rejection: TypeError: a is not a function. (In 'a()', 'a' is an instance of Object)
	(anonymous function) (70635.Zu5pSVmnIHw.js:1:124513)
	(anonymous function) (70635.Zu5pSVmnIHw.js:1:141396)
	l (app.9QvPhNMccVE.js:2:337446)
	l
	promiseReactionJob
[Error] Error loading page – "lovelace"
       TypeError: r is not a function. (In 'r()', 'r' is an instance of Object)
              (anonymous function) — show-edit-view-dialog.ts:43
              (anonymous function) — ha-panel-lovelace.ts:54
              l — bootstrap:19
              l
              promiseReactionJob
	(anonymous function) (app.9QvPhNMccVE.js:2:98247)
	promiseReactionJob
[Error] Unhandled Promise Rejection: TypeError: i is not a function. (In 'i()', 'i' is an instance of Object)
	dispatchException (app.9QvPhNMccVE.js:2:274590)
	(anonymous function) (app.9QvPhNMccVE.js:2:271682)
	r (app.9QvPhNMccVE.js:2:256651)
	u (app.9QvPhNMccVE.js:2:256891)
	promiseReactionJob
        https://local.ha.munro.cc/frontend_es5/app.9QvPhNMccVE.js
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (bootstrap, line 0)
```

I noticed the links shown to the right of the errors are not copied so here is a screenshot of the console:
<img width="772" alt="Screenshot 2024-05-04 at 3 56 37 PM" src="https://github.com/home-assistant/frontend/assets/1166024/9b0e4a6c-b264-473d-a86e-97aa4eb9ce3e">

**Comment by jaymunro** [comment]
@karwosts note there has also been one report from an Android user https://github.com/home-assistant/frontend/issues/20697#issuecomment-2092114465

**Comment by dimkin-eu** [comment]
Same here for old iPad with old ios 12.5.7 with old app version

**Comment by jeremychan** [comment]
It seems the problem lies in line 43 of show-edit-view-dialog.ts 

![image](https://github.com/home-assistant/frontend/assets/913979/85733c6a-2b10-4027-915f-d011912f7fbb)

which correspond to this line in the minified file 28316.Swg859rYwb0.js

![image](https://github.com/home-assistant/frontend/assets/913979/1fd81dd5-9e5f-453d-b674-b9db12b4d879)

in the debugger r here is an Object not a callable function

![image](https://github.com/home-assistant/frontend/assets/913979/deb7ea16-ef06-4e9f-847f-a11447623492)

**Comment by mheiker** [comment]
same to me

**Comment by apeiron69** [comment]
I'm sorry to have opened a new post about this issue. Just to say I have the same issue on iPad with 13.5.7 iOS and 2024.5.1 HA core installed

**Comment by apeiron69** [comment]
> Same here for HA 2024.5.1
> 
>     * old iPad
> 
>     * with old ios 12.5.7
> 
>     * with old app version
> 
>     * or with old chrome version
> 
> 
> reverted to 2024.4.4

How da I revert to 2924.4.x? Thank you

**Comment by silamon** [comment]
If you have this issue on iOS, just don't add more comments but use the thumbs up on the relevant comment perhaps the top comment.   

I can't debug or bisect the issue to see where it got introduced, but doing the same as Jeremy brings me to #20543.

**Comment by ljsquare** [comment]
I've got the same problem met iPad mini 2 iOS 12.5.7 and HA 2024.5.1

**Comment by piitaya** [comment]
I reproduced the issue in my dev env. Here's the unminified error :
```
VM2530:5 Unhandled Promise Rejection: TypeError: _regeneratorRuntime is not a function. (In '_regeneratorRuntime()', '_regeneratorRuntime' is an instance of Object)dispatchException @ VM2530:5(anonymous function) @ VM2530:179asyncGeneratorStep @ VM2538:11_throw @ VM2538:33promiseReactionJob @ [native code]:1
VM2530:5 Unhandled Promise Rejection: TypeError: _regeneratorRuntime is not a function. (In '_regeneratorRuntime()', '_regeneratorRuntime' is an instance of Object)dispatchException @ VM2530:5(anonymous function) @ VM2530:179asyncGeneratorStep @ VM2538:11_throw @ VM2538:33promiseReactionJob @ [native code]:1
VM2922:5 Unhandled Promise Rejection: TypeError: _regeneratorRuntime is not a function. (In '_regeneratorRuntime()', '_regeneratorRuntime' is an instance of Object)
```

I pushed a PR to quickly fix in the next patch release.

**Comment by prodato-alexander** [comment]
Same (or similar) here for Android 6.0:
Logger: frontend.js.es5.202405010
Source: components/system_log/__init__.py:333
First occurred: 11:25:35 AM (6 occurrences)
Last logged: 12:19:43 PM

Uncaught error from Chrome WebView 55.0.2883.91 on Android 6.0 TypeError: document.getElementsByTagName(...)[0].computedStyleMap is not a function :1:43

**Comment by dward32** [comment]
Has this now been fixed.. I rolled back to previous version, what do I do now.. ? install 2024.5.1 again?

**Comment by spacegaier** [comment]
> Has this now been fixed.. I rolled back to previous version, what do I do now.. ? install 2024.5.1 again?

It will be in the next patch release, so 2024.5.2

**Comment by steverep** [comment]
> I reproduced the issue in my dev env. Here's the unminified error :

@piitaya what did you run it on?

**Comment by piitaya** [comment]
@steverep On iPad Pro 12.9 2018 on iOS 12.

**Comment by steverep** [comment]
> @steverep On iPad Pro 12.9 2018 on iOS 12.

I cannot seem to repro yet and don't have older iOS hardware.  If you run on v17 but force legacy build (by editing index.html), does the problem still occur?

**Comment by steverep** [comment]
> I cannot seem to repro yet and don't have older iOS hardware. If you run on v17 but force legacy build (by editing index.html), does the problem still occur?

@piitaya nevermind - I can repro now and actual fix coming shortly.

**Comment by dimkin-eu** [comment]
did it managed to be included in 2024.5.2?
cant see something here https://www.home-assistant.io/blog/2024/05/01/release-20245/#release-202452---may-6

**Comment by Olsenoli** [comment]
Hi everyone, after the patch 2024.5.2 it works with old ipad and old ios.

**Comment by jaymunro** [comment]
Confirmed with iOS 17.4.1

**Comment by dimkin-eu** [comment]
confirmed with 12.5.7

**Comment by PaulHam211** [comment]
Anyones app crashing when navigating to the profile page (bottom left, profile photo icon)

**Comment by jaymunro** [comment]
> Anyones app crashing when navigating to the profile page (bottom left, profile photo icon). Edit: waiting on the profile page does eventually let you to scroll. but its very slow. something is wrong

All ok on both new and old iPads for me. Open a new issue if it still exists after the usual local suspects (cache, user, browser type, etc.) are confirmed not to be the issue.

**Comment by dward32** [comment]
all sorted for me, old iPad and IOS 12.5.7 - Thank you

**Comment by deltazerorsan** [comment]
> confirmed with iOS 12.5.7

Thanks!

Back again!

**Comment by benftrex** [comment]
I'm still having the same issues with the latest 2024.5.2. 
In my case the issue occurs with ANY browser (firefox,safari and chrome) that i use with an old imac 27" running Monterey (12.7.4). 
 I Have even tried the official hass app for mac here, and get the same errors:
I can correctly see all the settings, automations etc... But can't see any page with any card ! Where i get: "Error while loading page lovelace."

Everything works fine with my M1 Macbook running Sonoma, with all browsers. 
So it's definitely something to do with older macs or ios devices.

**Comment by computerfreek274** [comment]
It's back! In ver 
Core
2024.6.0b6
Supervisor
2024.06.0

**Comment by jaymunro** [comment]
> It's back! In ver Core 2024.6.0b6 Supervisor 2024.06.0

Can you open a new issue @computerfreek274 ?

**Comment by jaymunro** [comment]
> > It's back! In ver Core 2024.6.0b6 Supervisor 2024.06.0
> 
> Can you open a new issue @computerfreek274 ?

Reference this issue with a # and the number 20697 so other people can easily find it if they experience the same.
